src/parser/sparqlParser/SparqlQleverVisitor.cpp

```diff
@@ -2540,7 +2540,17 @@ ExpressionPtr Visitor::visitExists(Parser::GroupGraphPatternContext* pattern,
   auto group = visit(pattern);
   ParsedQuery argumentOfExists =
       std::exchange(parsedQuery_, std::move(queryBackup));
-  argumentOfExists.selectClause().setAsterisk();
+  SelectClause& selectClause = argumentOfExists.selectClause();
+  selectClause.setAsterisk();
+  for (const Variable& variable : visibleVariables_) {
+    // If we're not negated, we can safely add all visible variables to the
+    // expression, resulting in this filter to be optimized away if no variables
+    // match in the main pattern. For the negative case a variable is a wildcard
+    // match, so we just ignore this variable.
+    if (!negate || ad_utility::contains(visibleVariablesBackup, variable)) {
+      selectClause.addVisibleVariable(variable);
+    }
+  }
```

Comment:
This is somewhat approximate,as
1. Relevant variables might also be defined after the FILTER (Filters are technically applied last in a group). This might actually lead to wrong results if the query is at the beginning.
2. We could only consider variables from the same group, not from outside it, but that only affects the query plan and not the correctness.

Reply:
At least add a comment here.

Reply:
Or even better, we anyway 
have the position where we apply the language filters etc. after a complete group has been dealt with, can't we move these checks there (of course you then have to peek into the expressions to get all existsExpressions, but that should be doable).